debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleDatabaseSchema.java

```diff
@@ -106,4 +117,63 @@ public boolean isStorageInitializationExecuted() {
     public boolean historyExists() {
         return databaseHistory.exists();
     }
+
+    @Override
+    protected void removeSchema(TableId id) {
+        super.removeSchema(id);
+        lobColumnsByTableId.remove(id);
+    }
+
+    @Override
+    protected void buildAndRegisterSchema(Table table) {
+        if (getTableFilter().isIncluded(table.id())) {
+            super.buildAndRegisterSchema(table);
+
+            // Cache LOB column mappings for performance
+            buildAndRegisterTableLobColumns(table);
+        }
+    }
+
+    /**
+     * Get a list of large object (LOB) columns for the specified relational table identifier.
+     *
+     * @param id the relational table identifier
+     * @return a list of LOB columns, may be empty if the table has no LOB columns
+     */
+    public List<Column> getLobColumnsForTable(TableId id) {
+        return lobColumnsByTableId.getOrDefault(id, Collections.emptyList());
+    }
+
+    /**
+     * Returns whether the provided relational column model is a CLOB or NCLOB data type.
+     */
+    public static boolean isClobColumn(Column column) {
```

Comment:
Do we handle BLOBs and CLOBs separately somewhere? If not then `isLobColumn` method would be better and simplify the flow.

Reply:
We do this specifically in the `BaseChangeRecordEmitter#getReselectColumns` as we need to know if it's CLOB or BLOB to resolve which of the unavailable placeholder values types we should compare against to know whether we should reselect a column or not.  I could add an `isLobColumn` to satisfy the check at `LogMinerDmlParser` line 674.

Reply:
So I have an idea, at least to encapsulate this and re-use some behavior from `OracleValueConverters`.  This will allow me to change the visibility on `isClobColumn` and `isBlobColumn` to private.  Let me know what you think.